examples/react-native-project/package.json

```diff
@@ -0,0 +1,33 @@
+{
+  "name": "react-native-project",
+  "version": "0.0.0",
+  "license": "MPL-2.0",
+  "private": true,
+  "main": "./src/index.ts",
+  "scripts": {
+    "start": "expo start",
+    "android": "expo start --android",
+    "ios": "expo start --ios",
+    "web": "expo start --web",
+    "eject": "expo eject",
+    "config:generate": "app-config generate"
+  },
+  "dependencies": {
+    "@lcdev/app-config": "2 || 2.0.0-rc.3",
+    "expo": "39",
+    "expo-status-bar": "1",
+    "react": "16",
+    "react-dom": "16",
+    "react-native": "https://github.com/expo/react-native/archive/sdk-39.0.4.tar.gz",
+    "react-native-web": "0.13"
+  },
+  "devDependencies": {
+    "@babel/core": "7",
+    "@lcdev/react-native-app-config-transformer": "2 || 2.0.0-dev",
```

Comment:
won't this resolve incorrectly in the monorepo?

Reply:
You're right. Currently it resolves via the actual registry. We could use a `file://` dep path as an alternative.

Reply:
Does yarn correctly do `link:../app-config-react-...`? That'd be preferable.

Reply:
It does, though `link:` doesn't resolve transient dependencies. I gave it a try anyway but unfortunately failed.

Looks like [Yarn V2 supports a new protocol `portal:`](https://yarnpkg.com/features/protocols#whats-the-difference-between-link-and-portal) which behaves the same as `link:` but does resolve transient dependencies. Though most people will not have Yarn v2 installed globally...

Perhaps `file:` is the best option for now?

Reply:
Only thing would be if `file:` copies files before built files are made (as in, on a clean clone, the js files won't exist when yarn is installing / copying files)

Reply:
If it works, I'm all :+1:

Reply:
After testing, no `file:` doesn't ensure that the package is built before copy unfortunately. The best solution I've come up with is to add a `preinstall` hook that ensures the rest of the monorepo is install and built:

```json
{
  "scripts": {
    "preinstall": "cd ../../ && yarn install && yarn build"
  }
}
```

Let me know your thoughts. My biggest concern with all of this is the (likely) chance of somebody copying this RN example project as their base, and then immediately running into issues with the above `preinstall` script and `file:` dependency. This is actually technically true for all example projects, since many of their dev dependencies are only referenced in the root `package.json`.

Reply:
Yeah, agreed on all points here. I'm tempted to play around with `nohoist` again when I have some time. All options are kinda crappy.

Reply:
Sounds good. I've added the `preinstall` script for now.